packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -43,13 +47,26 @@ function adoptChildNode(node: Node, fn: ShadowRootResolver, shadowToken: string
 function initPortalObserver() {
     return new MO(mutations => {
         forEach.call(mutations, mutation => {
-            const { target: elm, addedNodes } = mutation;
+            const { target: elm, addedNodes, removedNodes } = mutation;
             // the target of the mutation should always have a ShadowRootResolver attached to it
             const fn = getShadowRootResolver(elm) as ShadowRootResolver;
             const shadowToken = getShadowToken(elm);
+
+            // First lets process removals in case we are in a case of removed and reinserted elements
+            for (let i = 0, len = removedNodes.length; i < len; i += 1) {
+                const node: Node = removedNodes[i];
+                if (
+                    !(compareDocumentPosition.call(elm, node) & Node.DOCUMENT_POSITION_CONTAINED_BY)
```

Comment:
you should be able to use `elm.isConnected` which is, I believe, polyfilled for IE11.

Reply:
We do not polyfill Node.isConnected. If you are planning to use it, would be a good time do it. There is an isConnected() method in [synthetic-shadow/src/env/node.ts](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/synthetic-shadow/src/env/node.ts#L78). Add that property to Node.prototype in faux-shadow/node.ts

Reply:
but is connected to where? this is the only pace that we have to clear the ownership, what happens if i keep a reference to a node, and remove it from the shadow, then (ahead in time) i added to the document? in this case the node will have the ownership, and wont work properly.

This observers, checks for mutations on `elm`:
1. in the case of removal, we can prune a `node` routine  if we know is contained by `elm`.
2. in the case of addding, we can prune a `node` routine if we know is not contained by `elm`.

but other than that, i dont think we should do assumptions about what the developer did or will do.